addons/sourcemod/scripting/surftimer/replay.sp

```diff
@@ -177,16 +185,37 @@ public void SaveRecording(int client, int zgroup, int style)
 	char szName[MAX_NAME_LENGTH];
 	GetClientName(client, szName, MAX_NAME_LENGTH);
 
+	//Add pre
+	int startFrame = g_iStartPressTick[client];
+	int endFrame = g_iRecordedTicks[client];
+
 	FileHeader header;
 	header.BinaryFormatVersion = BINARY_FORMAT_VERSION;
 	strcopy(header.Time, sizeof(FileHeader::Time), g_szFinalTime[client]);
-	header.TickCount = g_iRecordedTicks[client];
+	header.TickCount = endFrame - startFrame; //Add pre
 	strcopy(header.Playername, sizeof(FileHeader::Playername), szName);
 	header.Checkpoints = 0;
-	header.Frames = g_aRecording[client];
+
+	// Copy pasta stage separation method for proper Map/Bonus start frame
+	header.Frames = new ArrayList(sizeof(frame_t));
+	any aFrameData[sizeof(frame_t)];
+
+	for (int i = startFrame; i < endFrame; i++)
+	{
+		if (i == -1)
+		{
+			LogError("Map record cannot be saved. Client: \"%L\", startFrame: %d, endFrame: %d (g_iRecordedTicks: %d), i: %d, Path/File: %s", client, startFrame, endFrame, g_iRecordedTicks[client], i, sPath2);
```

Comment:
@Ciallo-Ani do you mean this?

Reply:
no, i mean `startframe` - `your_pre_seconds`

Reply:
For maps and bonuses that gets the proper value set in `OnStartTimerPress` and for stages in `Stages_StartRecording`